docs/DeveloperGuide.md

```diff
@@ -283,31 +283,112 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `TutorAid` and the **Actor** is the `Tutor`, unless specified otherwise)
 
-**Use case: Delete a person**
+**Use case 4: Update a student's progress**
 
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1. Tutor views the list of his/her students
+2. TutorAid displays the list of students
+3. Tutor identifies the student index to update their progress
+4. Tutor updates progress for that student
+5. TutorAid registers the new progress for the student at the specified index number
 
     Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty.
+* 1a. Invalid list command syntax
+
+    * 1a1. TutorAid displays an error message and requests the tutor to re-enter the command.
+
+    * 1a2. Tutor re-enters the command.
+
+      Steps 1a1-1a2 are repeated until the syntax entered is correct. Use case resumes from step 2.
+
+* 1b. The list is empty.
 
   Use case ends.
 
-* 3a. The given index is invalid.
+* 4a. Invalid add/delete progress syntax
+
+    * 4a1. TutorAid displays an error message and requests the tutor to re-enter the command.
+
+    * 4a2. Tutor re-enters the command
+
+      Steps 3a1 - 3a2 are repeated until a valid syntax is given. Use case resumes from step 5
+  
+* 5a. Incorrectly updating the progress of another student
```

Comment:
Hi, I was wondering if extension 5a is needed? This is because extensions usually describe exceptional/alternative flow of events, where there is a disruption when going through the steps stated in the MSS. But for 5a the tutor will technically be able to complete the whole MSS although the progress is being added to the wrong student?

Reply:
I think for this one I followed another use case so for now I think we keep it like this then we discuss later?